nibabel/nifti2.py

```diff
@@ -221,6 +221,17 @@ def _chk_eol_check(hdr, fix=False):
             rep.fix_msg = 'setting EOL check to 13, 10, 26, 10'
         return hdr, rep
 
+    @classmethod
+    def may_contain_header(klass, binaryblock):
```

Comment:
Maybe have `_sizeof_hdr_val` as class attribute, and share this code between Nifti1 and Nifti2?